Attempt to fix a random crash in CCharacterCore::Write() when connect…

pelya opened the pull request:
…ing to a server for the first time

Siile commented:
Hey, found it by accident and fixed with today's update: https://github.com/Siile/Ninslash/commit/30961cfb6af2a4aef8efe97cf3131aaffa9ff947#diff-c41ff3a713742e1dae8932366f246bc9R28

Let me know if it still crashes on server join.

pelya commented:
I've updated Android version to 0.2.6 two hours ago, and got this crash, so
probably it's not fixed yet.

It usually happens after starting the game and connecting to the server
with an active game going on.

If I start local server and connect to it before joinig other server, crash
mever happens, so I think it's some uninitialized data structure.

Applying this fix changes net hash value, so I cannot test with official
servers.

On May 14, 2017 1:22 AM, "Siile" <notifications@github.com> wrote:

> Hey, found it by accident and fixed with today's update: Siile/Ninslash@
> 30961cf#diff-c41ff3a713742e1dae8932366f246bc9R28
> <https://github.com/Siile/Ninslash/commit/30961cfb6af2a4aef8efe97cf3131aaffa9ff947#diff-c41ff3a713742e1dae8932366f246bc9R28>
>
> Let me know if it still crashes on server join.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Siile/Ninslash/pull/40#issuecomment-301278415>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AAJewNYVnOSCMNxnCYPYuyvbRf4UU6Njks5r5i0UgaJpZM4NaNKr>
> .
>

pelya commented:
Great! Let's see if it will help.

On May 22, 2017 8:49 PM, "Siile" <notifications@github.com> wrote:

> Merged #40 <https://github.com/Siile/Ninslash/pull/40>.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Siile/Ninslash/pull/40#event-1092266513>, or mute the
> thread
> <https://github.com/notifications/unsubscribe-auth/AAJewJl7iMhWqAljnxeml1gcK3ylrQ_-ks5r8cq1gaJpZM4NaNKr>
> .
>
